lib/types/ip.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package common
+package types
```

Comment:
I think we should be looking to break up the "types" package (as it is it isn't much better than "common").

For example, all the IP stuff should probably in a package called `net` or `network`.